x-pack/plugins/cases/common/types/domain/case/v1.ts

```diff
@@ -52,25 +52,6 @@ export const CaseSettingsRt = rt.strict({
   syncAlerts: rt.boolean,
 });
 
-export const customFieldValue = <C extends rt.Mixed>(codec: C) =>
-  rt.strict({ value: rt.union([rt.array(codec), rt.null]) });
-
-const CustomFieldText = rt.strict({
-  key: rt.string,
-  type: CustomFieldTextTypeRt,
-  field: customFieldValue(rt.string),
-});
-
-export const CustomFieldToggle = rt.strict({
-  key: rt.string,
-  type: CustomFieldToggleTypeRt,
-  field: customFieldValue(rt.boolean),
-});
-
-export const CustomFieldRt = rt.union([CustomFieldText, CustomFieldToggle]);
-
-const CaseCustomFieldsRt = rt.array(CustomFieldRt);
```

Comment:
I had to move the types to `x-pack/plugins/cases/common/types/domain/user_action/custom_fields/v1.ts` to avoid circular dependencies between the case and the user actions domain types.